Changes to Task health and state (DCOS-11264)

MatApple opened the pull request:
Feature reference: [DCOS-11264](https://mesosphere.atlassian.net/browse/DCOS-11264)

Changes:
1. Split up Status and Health into separate columns
2. Health dots only apply to active tasks
3. Health dots should have tooltips that explain what they mean e.g. Healthy, Unhealthy, No health checks available
4. Completed task rows are visually greyed out
5. States that are errors or considered bad are colored red

The new table-row state contained in the directory `src/styles/content/tables/table-cell-inactive` will be moved to CNVS (once approved) where it will join table-cell-danger, table-cell-warning, table-cell-success and table-cell-selected.

Pod Instances Table:
![](https://cl.ly/0a0u1s1i2b2c/Image%202016-12-02%20at%202.34.37%20PM.png)

Task Table:
![](https://cl.ly/0E1m3N0L1b2y/Image%202016-12-02%20at%202.32.08%20PM.png)

Tooltips:
![](https://cl.ly/1i29212q2a37/Image%202016-12-02%20at%203.48.40%20PM.png)
![](https://cl.ly/2E301R3W3P0I/Image%202016-12-02%20at%203.49.24%20PM.png)

jfurrow commented on src/styles/content/tables/table-cell-inactive/styles.less:
```diff
@@ -0,0 +1,131 @@
+
+    tr.inactive td,
+    td.inactive {
+      .element-text-style(table-cell-inactive);
```
Let's not use these mixins; instead just write the actual CSS properties. It will reduce this down to just a few lines of code.

MatApple replied:
I matched the CNVS format for this. The goal is to remove this from here and merge into CNVS so I think we should leave this as is unless @ashenden suggests otherwise. Thoughts?

orlandohohmeier replied:
I'm with @jfurrow and would vote against using those mixins due to the named reasons. The CSS that we write should be in line with our needs not those of CNVS.

MatApple replied:
I'm in favor of the suggestion. The intention was to put this code in CNVS so in that case it would have to follow the best practices of CNVS. Making this change means we keep this code and maintain it ourselves.

jfurrow replied:
@MatApple I reduced it to the bare minimum in this commit (Rafa asked me to help out to keep the PR moving, I hope you don't mind): https://github.com/dcos/dcos-ui/pull/1480/commits/3b4b6bcb9feb23cae4ef0c15e59eac0091ddaf65.

We'll just remove it when it's released in CNVS, so we won't be maintaining it in both places.

@ashenden If you'd like you can copy the mixins and variables definitions from this commit: https://github.com/dcos/dcos-ui/pull/1480/commits/5526d450c2b04ea91987b84a90aa1e88930caa3b

jfurrow commented on src/styles/content/tables/styles.less:
```diff
@@ -11,6 +11,29 @@
+        }
+
+        .expanding-table-primary-cell.is-expandable:before {
+          border-left-color: @table-cell-inactive-color;
```
Let's write this as:
```less
.expanding-table-primary-cell {

  &.is-expandable {

    &:before {
      border-left-color: @table-cell-inactive-color;
    }
  }
}
```

MatApple replied:
done!

jfurrow commented on src/styles/content/tables/table-cell-inactive/variables.less:
```diff
@@ -0,0 +1,373 @@
+
+// Normal
+
+@table-cell-inactive-color: color-lighten(@black, 75);
```
Without the mixins, this would be only 13 variables :)

rcorral replied:
can't we use the mixins to generate this?

MatApple replied:
see https://github.com/dcos/dcos-ui/pull/1480#discussion_r90745837

jfurrow commented on plugins/services/src/js/containers/pod-instances/PodInstancesTable.js:
```diff
@@ -363,16 +374,40 @@ class PodInstancesTable extends React.Component {
+        flex-direction-top-to-bottom">
+        <div className="table-cell-icon">
+          <Tooltip anchor="center" content={tooltipContent}>
+            <span className={status.dotClassName}></span>
```
We also need to add `flush` to this `span`, otherwise the tooltip won't be centered properly.

Also let's make it a self-closing element, e.g. `<span />`

MatApple replied:
done!

jfurrow commented:
Nice! This might be for a different PR, but the status dots look different for tasks/pods. One has a thin stroke while the other is filled. Do we know which is correct?

![](https://cl.ly/343W0b0M372s/Screen%20Shot%202016-12-02%20at%204.13.35%20PM.png)![](https://cl.ly/04100b0B0i12/Screen%20Shot%202016-12-02%20at%204.13.27%20PM.png)

rcorral commented:
Awesome!

I think we should make the health/status columns hide together.
![gif](https://cl.ly/231t301D0x0d/Screen%20Recording%202016-12-02%20at%2004.42%20PM.gif)

rcorral commented:
All of the tooltips say "Healthy", I think it's because we cache the column.
![img](https://cl.ly/340Z353h3T2Y/Image%202016-12-02%20at%204.43.26%20PM.png)

MatApple commented:
@jfurrow - addressed all comments. The health dots with just a border will be present on tasks that are in a completed state (i.e. killed, complete, error, etc). A grey health dot with fill represents a task that is still active but has no health checks.

@rcorral - fixed the caching and hiding of health/status so they are in sync.

jfurrow commented:
@MatApple Thanks for the explanation! It looks great, good catch!!!

ashenden commented:
FWIW, I suggested this additional row state be a PR against CNVS not dcos-UI.    
  
  

  
  
  

  
  
>   
> On Dec 3, 2016 at 11:05 AM,  <Orlando Hohmeier (mailto:notifications@github.com)>  wrote:
>   
>   
>   
>
> @orlandohohmeier  commented on this pull request.
>
>     
>
> In  src/styles/content/tables/table-cell-inactive/styles.less (https://github.com/dcos/dcos-ui/pull/1480):
>
>   
>   >  @@ -0,0 +1,131 @@ +&  when (@table-cell-inactive-enabled) { + /* Table */ + + .table { + + tr.inactive td, + td.inactive { + .element-text-style(table-cell-inactive);  
>   
>
> I'm with  @jfurrow (https://github.com/jfurrow)  and would vote against using those mixins due to the named reasons. The CSS that we write should be in line with our needs not those of CNVS.
>
>   
>
> —
>  You are receiving this because you were mentioned.
>  Reply to this email directly,  view it on GitHub (https://github.com/dcos/dcos-ui/pull/1480), or   mute the thread (https://github.com/notifications/unsubscribe-auth/AARwqyvMfHc1BREJBS--rRkEcMXj1vaYks5rEb1wgaJpZM4LDGLS).
>
>   
>   
>     
>   
>   
>

MatApple commented:
This PR also fixes a bug where active PODs weren't previously shown on the "active" filter selection. That also meant that active PODs reported the wrong health status. Fixed this here https://github.com/dcos/dcos-ui/pull/1480/files#diff-ff5234fa70f403ffc708e3774cac23a9R27

MatApple commented:
@ashenden  cc/ @jfurrow, @rcorral, @orlandohohmeier  that was the intention. I wanted to get it merged here so that it works and is testable for reviewers before migrating it to a CNVS PR.

jfurrow commented:
I just noticed the table is displaying strangely for me, but it's not from this PR. I'll fix in master.
![](https://cl.ly/0G1a2R2Q300t/Screen%20Shot%202016-12-03%20at%201.19.26%20PM.png)

jfurrow commented:
retest this please
